src/ReaderThemeSupportFeatures.php

```diff
@@ -0,0 +1,376 @@
+<?php
+/**
+ * Class ReaderThemeSupportFeatures.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use AMP_Options_Manager;
+
+/**
+ * Stores the primary theme's theme support features when Reader template mode is active and then adds the necessary styles to support them.
+ *
+ * Note that this service is not conditional based on whether Reader mode is selected because it needs to run logic
+ * when transitioning from non-reader to Reader mode.
+ *
+ * @package AmpProject\AmpWP
+ * @since 2.1
+ * @internal
+ */
+final class ReaderThemeSupportFeatures implements Service, Registerable {
+
+	/**
+	 * Theme feature slug for editor-color-palette.
+	 *
+	 * @var string
+	 */
+	const FEATURE_EDITOR_COLOR_PALETTE = 'editor-color-palette';
+
+	/**
+	 * Theme feature slug for editor-gradient-presets.
+	 *
+	 * @var string
+	 */
+	const FEATURE_EDITOR_GRADIENT_PRESETS = 'editor-gradient-presets';
+
+	/**
+	 * Theme feature slug for editor-font-sizes.
+	 *
+	 * @var string
+	 */
+	const FEATURE_EDITOR_FONT_SIZES = 'editor-font-sizes';
+
+	/**
+	 * Key slug.
+	 *
+	 * @var string
+	 */
+	const KEY_SLUG = 'slug';
+
+	/**
+	 * Key size.
+	 *
+	 * @var string
+	 */
+	const KEY_SIZE = 'size';
+
+	/**
+	 * Key color.
+	 *
+	 * @var string
+	 */
+	const KEY_COLOR = 'color';
+
+	/**
+	 * Key gradient.
+	 *
+	 * @var string
+	 */
+	const KEY_GRADIENT = 'gradient';
+
+	/**
+	 * Supported features.
+	 *
+	 * @var array[]
+	 */
+	const SUPPORTED_FEATURES = [
+		self::FEATURE_EDITOR_COLOR_PALETTE    => [ self::KEY_SLUG, self::KEY_COLOR ],
+		self::FEATURE_EDITOR_GRADIENT_PRESETS => [ self::KEY_SLUG, self::KEY_GRADIENT ],
+		self::FEATURE_EDITOR_FONT_SIZES       => [ self::KEY_SLUG, self::KEY_SIZE ],
+	];
+
+	/**
+	 * Reader theme loader.
+	 *
+	 * @var ReaderThemeLoader
+	 */
+	private $reader_theme_loader;
+
+	/**
+	 * ReaderThemeLoader constructor.
+	 *
+	 * @param ReaderThemeLoader $reader_theme_loader Reader theme loader.
+	 */
+	public function __construct( ReaderThemeLoader $reader_theme_loader ) {
+		$this->reader_theme_loader = $reader_theme_loader;
+	}
+
+	/**
+	 * Register the service with the system.
+	 *
+	 * @return void
+	 */
+	public function register() {
+		add_filter( 'amp_options_updating', [ $this, 'filter_amp_options_updating' ] );
+		add_action( 'after_switch_theme', [ $this, 'update_cached_theme_support' ] );
```

Comment:
This doesn't work when switching the theme with WP-CLI.

Reply:
Now it works with b62a8c873.

Reply:
Note that when you switch a theme with WP-CLI, the `after_switch_theme` action will only fire the next time that a user accesses the site.